Block/Catalog/Product/Usage.php

```diff
@@ -490,4 +519,166 @@ public function getCustomLicenseId()
             \Magento\Store\Model\ScopeInterface::SCOPE_STORE
         );
     }
+
+    /**
+     * @return mixed
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function getPreviousSelectHtml()
+    {
+        $extraParams = '';
+        $select = $this->getLayout()->createBlock(
+            \Magento\Framework\View\Element\Html\Select::class
+        )->setData(
+            [
+                'id' => 'usage_previous_usages',
+                'class' => 'required product-custom-option admin__control-select usage-select-box'
+            ]
+        );
+
+        $select->setName('previous_category')->addOption('', __('-- Please Select --'));
+
+        foreach ($this->getPreviousCategories() as $category) {
+
+            $select->addOption(
+                $category['id'],
+                $category['name']
+            );
+        }
+        $extraParams .= ' data-selector="' . $select->getName() . '"';
+        $select->setExtraParams($extraParams);
+
+        return $select->getHtml();
+    }
+
+    /**
+     * @return array
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getPreviousCategories()
+    {
+        $previousCategories = [];
+        $items = $this->checkoutSession->getQuote()->getAllVisibleItems();
+
+        foreach ($items as $item) {
+            /**
+             * @var \Magento\Quote\Model\Quote\Item $item
+             */
+            $id = '';
+            $value = '';
+            $previousCategoriesByItems = [];
+            $getBuyRequest = $item->getBuyRequest();
+            foreach ($getBuyRequest as $data) {
+                if (isset($data['usage_category'])) {
+                    $id .= $data['usage_category'] . ' - ';
+                    $value .= $this->getCategoryName($data['usage_category']) . ' - ';
+                    $id .= $data['usage_id'][$data['usage_category']] . ' - ';
+                    $value .= $this->getUsageName($data['usage_id'][$data['usage_category']]) . ' - ';
+                    if (isset($data['options'])) {
+                        foreach ($data['options'] as $key => $option) {
+                            $id .= $key . ':' . $option . ' - ';
+                            $value .= $this->getOptionName($option) . ' - ';
+                        }
+                    }
+                }
+            }
+            if (!empty($id) && !empty($value)) {
+                $previousCategoriesByItems['id'] = rtrim(ltrim($id, ' - '), ' - ');
```

Comment:
One `trim()` should do it and be more readable/faster and the second parameter is a character mask so ` -` is enough.
```suggestion
                $previousCategoriesByItems['id'] = trim($id, ' -');
```